src/protocol/boolean/prefix_or.rs

```diff
@@ -361,40 +390,46 @@ mod tests {
                 acc | x
             });
 
-            // Generate secret shares
-            #[allow(clippy::type_complexity)]
-            let (s0, (s1, s2)): (
-                Vec<Replicated<Fp2>>,
-                (Vec<Replicated<Fp2>>, Vec<Replicated<Fp2>>),
-            ) = input
-                .iter()
-                .map(|&x| {
-                    let y = share(x, &mut rand);
-                    (y[0], (y[1], y[2]))
-                })
-                .unzip();
-
-            // Execute
-            let pre0 = PrefixOr::new(&s0);
-            let pre1 = PrefixOr::new(&s1);
-            let pre2 = PrefixOr::new(&s2);
-            let iteration = format!("{}", i);
-            let result = try_join_all(vec![
-                pre0.execute(ctx[0].narrow(&iteration), RecordId::from(0_u32)),
-                pre1.execute(ctx[1].narrow(&iteration), RecordId::from(0_u32)),
-                pre2.execute(ctx[2].narrow(&iteration), RecordId::from(0_u32)),
-            ])
-            .await
-            .unwrap();
+            let result = prefix_or(&input).await?;
 
             // Verify
-            assert_eq!(input.len(), result[0].len());
-            for (j, &e) in expected.iter().enumerate().take(input.len()) {
-                assert_eq!(
-                    e,
-                    validate_and_reconstruct((result[0][j], result[1][j], result[2][j])),
-                );
-            }
+            assert_eq!(expected.len(), result.len());
+            zip(expected, result).for_each(|(e, r)| assert_eq!(e, r));
         }
+
+        Ok(())
+    }
+
+    #[tokio::test]
+    /// Test PrefixOr with the input ⊆ F_p (i.e. Fp31)
+    pub async fn fp31() -> Result<(), BoxError> {
+        let mut rng = rand::thread_rng();
+
+        // Test 32-bit bitwise shares with randomly distributed bits, for 16 times.
+        // The probability of i'th bit being 0 is 1/2^i, so this test covers inputs
+        // that have all 0's in 5 first bits.
+        for _ in 0..TEST_TRIES {
+            let len = BITS;
+            // Generate a vector of Fp31::ZERO or Fp31::ONE from randomly picked bool values
+            let input: Vec<Fp31> = (0..len)
+                .map(|_| Fp31::from(u128::from(rng.gen::<bool>())))
+                .collect();
+            let mut expected: Vec<Fp31> = Vec::with_capacity(len);
+
+            // Calculate Prefix-Or of the secret number
+            input.iter().fold(0, |acc, &x| {
+                let sum = acc + x.as_u128();
```

Comment:
This will wrap around if you attempt this on a sufficiently long number. For example, a 64 bit number will (with high probability) have over 31 ones.